TombEngine/Sound/sound.cpp

```diff
@@ -146,44 +149,46 @@ bool LoadSample(char* pointer, int compSize, int uncompSize, int index)
 	return true;
 }
 
-bool SoundEffect(int effectID, Pose* position, SoundEnvironment condition, float pitchMultiplier, float gainMultiplier)
+// TODO: Use std::optional for pose argument.
+bool SoundEffect(int soundID, Pose* pose, SoundEnvironment soundEnv, float pitchMult, float gainMult)
 {
 	if (!g_Configuration.EnableSound)
 		return false;
 
-	if (effectID >= g_Level.SoundMap.size())
+	if (soundID >= g_Level.SoundMap.size())
 		return false;
 
 	if (BASS_GetDevice() == -1)
 		return false;
 
-	if (condition != SoundEnvironment::Always)
+	// Test if sound effect environment matches camera environment.
+	if (soundEnv != SoundEnvironment::Always)
 	{
-		// Get current camera room's environment
-		auto cameraCondition = TestEnvironment(ENV_FLAG_WATER, Camera.pos.RoomNumber) ? SoundEnvironment::Water : SoundEnvironment::Land;
-
-		// Don't play effect if effect's environment isn't the same as camera position's environment
-		if (condition != cameraCondition)
+		bool isCameraUnderwater = TestEnvironment(ENV_FLAG_WATER, Camera.pos.RoomNumber);
```

Comment:
Shouldn't you add similar condition for swamp environment type?

Reply:
The `Underwater` sound condition is unique in that sounds will only play only when the camera is submerged. The other conditions are "above water".

Reply:
Yes, that's what I mean. If object is submerged into the swamp, shouldn't sounds also be canceler? For example, there is a funny bug when Lara dies in the swamp, you can hear her grunt.

Reply:
`Underwater` is unique in that sense. `Swamp` works more like `ShallowWater`.

Reply:
I get it then. Would playback condition be checked against `Underwater` case if sound is emitted within the swamp room then?